controller/share_manager_controller.go

```diff
@@ -430,8 +430,10 @@ func (c *ShareManagerController) detachShareManagerVolume(sm *longhorn.ShareMana
 	log := getLoggerForShareManager(c.logger, sm)
 
 	shareManagerAttachmentTicketID := longhorn.GetAttachmentTicketID(longhorn.AttacherTypeShareManagerController, sm.Name)
-	log.Infof("Removing volume attachment ticket: %v to detach the volume %v", shareManagerAttachmentTicketID, va.Name)
-	delete(va.Spec.AttachmentTickets, shareManagerAttachmentTicketID)
+	if _, ok := va.Spec.AttachmentTickets[shareManagerAttachmentTicketID]; ok {
+		log.Infof("Removing volume attachment ticket: %v to detach the volume %v", shareManagerAttachmentTicketID, va.Name)
```

Comment:
This is OK, but why there is a chance the `va.Spec.AttachmentTickets` does not have `shareManagerAttachmentTicketID`?

Reply:
If the share-manager volume is detached, `shareManagerAttachmentTicketID` is deleted.
The periodic check will print the message constantly.